lib/compat/wordpress-6.3/script-loader.php

```diff
@@ -83,3 +83,107 @@ function( $settings ) {
 	},
 	100
 );
+
+
+if ( ! wp_is_block_theme() ) {
+	// Load the block styles inline for non-block themes.
+	add_filter( 'should_load_separate_core_block_assets', '__return_true', 1 );
+	// Remove the block styles from the footer.
+	remove_action( 'wp_footer', 'wp_maybe_inline_styles', 1 );
+}
+
+/**
+ * Print block styles inline for non-block (classic) themes.
+ * Improves performance for classic themes by only loading the styles that are needed
+ * for the blocks that are actually used on the page, instead of loading all block styles.
+ * At the same time, this reduces layout shifts by inlining these styles
+ * instead of printing them all in the footer.
+ *
+ * NOTE: When backported to Core,
+ * this should probably be merged with the `wp_maybe_inline_styles` function.
+ *
+ * This function is hooked on `render_block`.
+ *
+ * @param string $block_content The block content.
+ * @param array  $block         The full block, including name and attributes.
+ *
+ * @return string The block content.
+ */
+function gutenberg_inline_block_styles_on_classic_themes( $block_content, $block ) {
+	// Bail early if not a classic theme.
+	if ( wp_is_block_theme() ) {
+		return $block_content;
+	}
+
+	// Get all registered blocks. Use a static var to avoid calling this
+	// for each block that gets rendered on the page.
+	static $all_registered_blocks;
+	if ( ! $all_registered_blocks ) {
+		$all_registered_blocks = WP_Block_Type_Registry::get_instance()->get_all_registered();
+	}
+
+	$style_handles = array();
+	if ( ! empty( $all_registered_blocks[ $block['blockName'] ] ) ) {
+		$style_handles = $all_registered_blocks[ $block['blockName'] ]->style_handles;
+	}
+
+	// Bail early if there are no styles for this block.
+	if ( empty( $style_handles ) ) {
+		return $block_content;
+	}
+
+	/* This filter is documented in wp-includes/script-loader.php */
+	$total_inline_limit = apply_filters( 'styles_inline_size_limit', 20000 );
+
+	// A static var to keep track of the total inlined CSS size.
+	static $total_inline_size = 0;
+
+	// Get the WP_Styles object.
+	$wp_styles = wp_styles();
+
+	foreach ( $style_handles as $style_handle ) {
+		// Skip stylesheet if it has already been printed.
+		if ( in_array( $style_handle, $wp_styles->done, true ) ) {
+			continue;
+		}
+
+		$path  = wp_styles()->get_data( $style_handle, 'path' );
+		$after = wp_styles()->get_data( $style_handle, 'after' );
+		if ( is_array( $after ) ) {
+			$after = implode( '', $after );
+		}
+
+		// Skip stylesheet if it has no path or "after" data.
+		if ( ! $path && ! $after ) {
+			continue;
+		}
+
+		// Get the CSS for this stylesheet.
+		$css = file_exists( $path ) ? file_get_contents( $path ) : '';
+
+		// Add the "after" styles.
+		$css .= $after;
+
+		// Get the size of the CSS.
+		$styles_size = strlen( $css );
```

Comment:
wp_maybe_inline_styles use filesize over doing this. `file_get_contents` can be very expensive. Let's use filesize here as well.

Reply:
I agree we should use `filesize()`. That said, we need to use `file_get_contents()` regardless, just to get the actual CSS to inline.